src/metabase/query_analysis/native_query_analyzer.clj

```diff
@@ -177,17 +177,39 @@
                          [:= :t.db_id db-id]
                          (into [:or] (map table-query tables))]}))))
 
+(defn- fill-missing-table-ids-hack
+  "See if we can qualify the schema and table-id for any explicit field refs which couldn't resolve their field"
+  [table-refs field-refs]
+  ;; Note, at this point we have given up on any pretense of respecting case sensitivity
+  (let [t->ids  (group-by (comp u/lower-case-en :table) table-refs)
+        s+t->id (u/index-by (juxt (comp u/lower-case-en :schema)
+                                  (comp u/lower-case-en :table))
+                            table-refs)]
+    (into (empty field-refs)
```

Comment:
why empty? is this a hidden `into []` case? :)

Reply:
Being agnostic of list, vec, or set makes it nicer if we need to make changes elsewhere in code. I'm proud not even to remember which it is here 😄